### ISSUE TITLE
Spec for sys.meta_path too strict

### ISSUE THREAD

**Issue by jaraco** [python/typeshed#2468]
In [this comment](https://github.com/python/mypy/issues/5589#issuecomment-419608778), we learned that typeshed is too constrained on its spec for `sys.meta_path`, expecting instances of MetaPathFinder, whereas the existing values are of `Type[MetaPathFinder]`, but more importantly, the objects there need not be instances of any particular class, but must conform to a protocol (having a `find_spec` or `find_module` on Python 3 and having a `find_module` on Python 2).

**Comment by pfmoore** [comment]
I've just hit this as well, [here](https://github.com/pfmoore/editables/pull/14#issuecomment-974091686). Is there a workaround until it's fixed?

**Comment by layday** [comment]
#6344

**Comment by Akuli** [comment]
I believe #5349 fixed this already, and mypy just uses an old version of typeshed.

**Comment by pfmoore** [comment]
> I believe #5349 fixed this already

Nope, see my review comment. The current code doesn't allow for class or static methods.

**Comment by pfmoore** [comment]
Is there any progress on this? I'm reviewing whether to do anything to add type annotations to `editables`, and this is a blocker for me.

**Comment by layday** [comment]
It looks like `find_module` and `invalidate_caches` were re-introduced in the typeshed at some point.

**Comment by layday** [comment]
Actually, I was looking at the wrong place (at`importlib.abc.MetaPathFinder`), and that hasn't changed. Can you explain what problem is it that you are experiencing?

**Comment by pfmoore** [comment]
Sorry, I can't. My comment that I mentioned [above](https://github.com/python/typeshed/issues/2468#issuecomment-974153244) seems to have been lost somehow.

I guess it'll have to wait until I get the motivation to try to apply types again, and I encounter the issue once more...

**Comment by abravalheri** [comment]
@jaraco would #11890 be enough to address this issue? (it does not handle 2.7, or `find_module` as it is deprecated).

**Comment by jaraco** [comment]
Yes, that sounds like a proper fix. Thank you!
